app/collaborator-detail.component.ts

```diff
@@ -61,4 +72,28 @@ export class CollaboratorDetailComponent implements OnInit {
     };
     fileReader.readAsDataURL(file);
   }
+
+  /**
+   * Met à jour les différentes propriétés à partir du collaborateur
+  **/
+  update() {
+    this.getManagers();
+    this.getManagedCollaborators();
+  }
+
+  ngOnChanges(changes: {[propName: string]: any}) {
+    //Lorsque l'on change la valeur de l'attribut "collaborator"
+    //on souhaite mettre à jour le composant
+    //Mais cette méthode est appelé également à la première initialisation
+    //Pour éviter un doublon d'appel, on ne met à jour le composant
+    //seulement si une valeur été initialisée auparavant
+    if (changes['collaborator'].previousValue != {}) {
+      this.update();
+    }
+  }
+
+  collabClicked(collaborator : Collaborator) {
```

Comment:
Après recherche sur le fonctionnement de ngOnChanges, il faut savoir que la modification d'une propriété dans le code ne déclenche pas le ngOnChanges. Seul les modifications via le template déclenche cette méthode.

http://www.bennadel.com/blog/3053-changing-directive-inputs-programmatically-won-t-trigger-ngonchanges-in-angularjs-2-beta-9.htm